aim/db/infra_model.py

```diff
@@ -154,3 +155,12 @@ class HostLinkNetworkLabel(model_base.Base, model_base.AttributeMixin):
     host_name = sa.Column(sa.String(128), primary_key=True)
     network_label = sa.Column(sa.String(64), primary_key=True)
     interface_name = sa.Column(sa.String(32), primary_key=True)
+
+
+class ApicAssignment(model_base.Base, model_base.AttributeMixin):
+    __tablename__ = 'aim_apic_assignment'
+
+    apic_host = sa.Column(sa.String(128), primary_key=True)
+    aim_aid_id = sa.Column(sa.String(64))
```

Comment:
Have you considered also making aim_aid_id a primary key? This would allow the situation where there are more AIDs than APICs to be represented in the DB. 

Then, when connecting, an AID could query for all the records, and pick the APIC that appears to be least loaded. Of course race conditions could result in all AIDs picking the same APIC, so some sort of re-check after connecting might make sense, or some sort of random periodic rebalancing.

Reply:
As explained in one of the earlier comment, the design in this patch is that we use the new table to indicate who has claimed the ownership of an APIC already instead of showing who is using which APIC so.